x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/index/store/SearchableSnapshotIndexInput.java

```diff
@@ -276,7 +296,7 @@ boolean canContinueSequentialRead(int part, long pos) {
 
         int read(byte[] b, int offset, int length) throws IOException {
             assert this.pos < maxPos : "should not try and read from a fully-read stream";
-            int read = inputStream.read(b, offset, length);
+            final int read = readFully(inputStream, b, offset, length, () -> {});
```

Comment:
I'm wondering whether we should throw an EOFException if this hits EOF too. Do we need to behave differently here?

Reply:
Oh no I get it now, we might use this to read past the end of a stream that we previously opened. Never mind.

Reply:
Yes